app/Http/Controllers/CustomerController.php

```diff
@@ -0,0 +1,275 @@
+
+            DB::commit();
+
+            return $this->sendResponse(new CustomerResource($customer), 'Customer created successfully.', 201);
+
+        } catch(Exception $e) {
+            
+            DB::rollBack();
+
+            return $this->sendError($e->getMessage(), [], 500);
+        }
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    
+    public function update(Request $request, $id)
+    {
+        DB::beginTransaction();
+        
+        try {
+
+            $input = $request->all();
+            $customer = Customer::find($id);
+
+            if (!$customer) {
+                return $this->sendError('Customer not found.'); 
+            }
+
+            $validator = Validator::make($input, $this->rules(), $this->fields());
+
+            if($validator->fails()){
+                return $this->sendError('Validation Error.', $validator->errors(), 400);   
+            }
+
+            if (request('profile_pic')) {
+                $input['profile_pic_path'] = $this->customerProfilePic($request);
+            }
+            
+            $customer->update($input);
+
+            DB::commit();
+
+            return $this->sendResponse(new CustomerResource($customer), 'Customer updated successfully.', 200);
+
+        } catch (Exception $e) {
+
+            DB::rollBack();
+
+            return $this->sendError($e->getMessage(), [], 500); 
+        }
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+
+    public function destroy($id)
+    {
+        DB::beginTransaction();
+
+        try {
+
+            $customer = Customer::find($id);
+
+            if (!$customer) {
+                return $this->sendError('Customer not found.'); 
+            }
+
+            $customer->delete();
+
+            DB::commit();
+            
+            return $this->sendResponse([], 'Customer deleted successfully.');
+
+        } catch (Exception $e) {
+            
+            DB::rollBack();
+
+            return $this->sendError($e->getMessage(), [], 500);
+        }
+    }
+
+    /**
+     * Display a list of the Customers with incomes
+     * 
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     * 
+     */
+
+    public function reportListing(Request $request)
+    {
+        try {
+
+            $input = $request->all();
+
+            $validator = Validator::make($input, $this->rulesForReportList(), $this->fields());
+
+            if($validator->fails()){
+                return $this->sendError('Validation Error.', $validator->errors(), 400);   
+            }
+
+            $startDate = isset($input['startDate'])? $input['startDate']: null;
+            $endDate = isset($input['endDate'])? $input['endDate']: null;
```

Comment:
we have 2 possible solutions to this:

```suggestion
            $startDate = $input['startDate'] ?? null;
            $endDate = $input['endDate'] ?? null;
```

```suggestion
            $startDate = $request->has('startDate') ? $input['starrtDate'] : null;
            $endDate = $request->has('endDate') ? $input['starrtDate'] : null;
```